numpy/core/tests/test_ufunc.py

```diff
@@ -2546,7 +2573,8 @@ def test_reduce_casterrors(offset):
     with pytest.raises(ValueError, match="invalid literal"):
         # This is an unsafe cast, but we currently always allow that.
         # Note that the double loop is picked, but the cast fails.
-        np.add.reduce(arr, dtype=np.intp, out=out)
+        # `initial=None` disables the use of an identity here.
```

Comment:
Is this just to ensure refcounts on it do not contribute?

Reply:
Expanding the comment, the point is that the cast error may happen while copying the "first values", but that only happens when there is no initial.